src/main/java/com/salesforce/sld/csrf/StatelessCSRFTokenManager.java

```diff
@@ -393,6 +393,14 @@ private boolean validateTokenInternal( String token, String sessionID, String...
             byte[] key = sessionID.getBytes( Charset.defaultCharset() );
 
             byte[] tokenByte = decodeToken( token );
+            if( tokenByte.length < TOKEN_SIZE )
+            {
+                String error = new StringBuffer().append("The format of the CSRF token received is incorrect")
```

Comment:
This doesn't need to be a StringBuffer as it's a static string. Also, don't use StringBuffer except when you're sharing the object among threads. StringBuffer is synchronized on each method. StringBuilder is the unsynchronized version with a significant speed boost